File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/persistence/cdmfolder/types/ConstantEntity.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.types;

import com.fasterxml.jackson.databind.JsonNode;
import java.util.List;

public class ConstantEntity {
    private String explanation;
    private String constantEntityName;
    private JsonNode entityShape;
    private List<List<String>> constantValues;

    public String getExplanation() {
        return this.explanation;
    }

    public void setExplanation(final String explanation) {
        this.explanation = explanation;
    }

    public String getConstantEntityName() {
        return this.constantEntityName;
    }

    public void setConstantEntityName(final String constantEntityName) {
        this.constantEntityName = constantEntityName;
    }

    public JsonNode getEntityShape() {
        return this.entityShape;
    }

    public void setEntityShape(final JsonNode entityShape) {
        this.entityShape = entityShape;
    }

    public List<List<String>> getConstantValues() {
        return this.constantValues;
    }

    public void setConstantValues(final List<List<String>> constantValues) {
        this.constantValues = constantValues;
    }
}

File: objectModel/Python/cdm/persistence/syms/types/constant_entity.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

from typing import Union, List

from cdm.utilities import JObject

from .entity_reference import EntityReference


class ConstantEntity(JObject):
    def __init__(self):
        super().__init__()

        self.explanation = None  # type: str
        self.constantEntityName = None  # type: str
        self.entityShape = None  # type: Union[str, EntityReference]
        self.constantValues = None  # type: List[List[str]]
